examples/darwin-framework-tool/commands/clusters/ReportCommandBridge.h

```diff
@@ -237,7 +276,88 @@ class SubscribeEvent : public ModelCommand {
     chip::Optional<bool> mKeepSubscriptions;
     chip::Optional<bool> mAutoResubscribe;
     chip::Optional<chip::EventNumber> mEventNumber;
+    chip::Optional<bool> mIsUrgent;
     bool mSubscriptionEstablished = NO;
     uint16_t mMinInterval;
     uint16_t mMaxInterval;
+
+    void Shutdown() override
+    {
+        mSubscriptionEstablished = NO;
+        ModelCommand::Shutdown();
+    }
+
+    bool DeferInteractiveCleanup() override { return mSubscriptionEstablished; }
+
+private:
+    chip::ClusterId mClusterId = chip::kInvalidAttributeId;
+    chip::EventId mEventId;
+};
+
+class ReadEvent : public ModelCommand {
+public:
+    ReadEvent()
+        : ModelCommand("read-event-by-id")
+    {
+        AddArgument("cluster-id", 0, UINT32_MAX, &mClusterId);
+        AddArgument("event-id", 0, UINT32_MAX, &mEventId);
+        AddArgument("event-min", 0, UINT64_MAX, &mEventNumber);
+        ModelCommand::AddArguments();
+    }
+
+    ReadEvent(chip::ClusterId clusterId)
+        : ModelCommand("read-event-by-id")
+        , mClusterId(clusterId)
+    {
+        AddArgument("event-id", 0, UINT32_MAX, &mEventId);
+        AddArgument("event-min", 0, UINT64_MAX, &mEventNumber);
+        ModelCommand::AddArguments();
+    }
+
+    ReadEvent(const char * _Nonnull eventName)
+        : ModelCommand("read-event")
+    {
+        AddArgument("event-name", eventName);
```

Comment:
Where does this set mClusterId or mEventId?  I'm not clear on how this constructor ends up working, at least to set the members we use below. I also don't see it actually being used in the codegen.

Reply:
All arguments including mClusterId and mEventId are set in InitArgument method of Command class which is one of the parent classes of ReadEvent. If user input of darwin-framework-tool is including read-event-by-id, ReadEvent instance generates and all member variables are set by initArgument method based on the user input before calling SendCommand of ReadEvent.

Reply:
"read-event" command is deleted and it will be implemented later if app developer has some needs (now only read-event-by-id is being used in our side).